src/parser/analyse/Story/children.ts

```diff
@@ -184,40 +195,22 @@ function findChildrenPropSnippetBlock(
  * <Component {...args } />
  * ```
  */
-function getSnippetBlockBodyRawCode(originalCode: string, node: SnippetBlock) {
+async function getSnippetBlockBodyRawCode(originalCode: string, node: SnippetBlock) {
   const { body } = node;
   const { nodes } = body;
   const firstNode = nodes[0];
   const lastNode = nodes[nodes.length - 1];
   const rawCode = originalCode.slice(firstNode.start, lastNode.end);
 
-  return sanitizeCodeSlice(rawCode);
+  return await prettifyCodeSlice(rawCode);
 }
 
-/**
- * WARN: The current solution was written very quickly. Expect bugs.
- * TODO:
- * Need to figure a safer way to remove unwanted leading and ending tabs, spaces, new lines and so on.
- */
-function sanitizeCodeSlice(rawCode: string): string {
-  return rawCode.replace(/(\n)/g, '').trim();
-}
-
-function getDefineMetaComponentName(svelteASTNodes: Params['svelteASTNodes']) {
-  const { component } = extractMetaPropertiesNodes({
-    nodes: svelteASTNodes,
-    properties: ['component'],
+async function prettifyCodeSlice(rawCode: string) {
+  /**
+   * FIXME: Perhaps we don't need to prettify the code at this point, and do it at runtime instead?
```

Comment:
I investigated this prior to your prettier implementation here, and I'm unsure.

It makes a lot of sense to do it at runtime, because it's there that we replace `{...args}` with (possibly) a lot of attibutes, so that will change the format. On the other hand, doing it at runtime means bundling in prettier in the final build output, and I don't think that's the right tradeoff. Prettier is a huge package.

I'm not even sure we should have it at all, here. Can you talk about what it does for us here? Is it all about removing indentation? If so, we should be able to do use `dedent` that we already use a couple of other places.

If users have formatted their files correctly, will it look good in our code viewer too? or is there more to it?

Reply:
Forget it. My bad. I over-engineered with this solution.

I completely forgot about the existence of `dedent`, and that it could solve my problem with removal of leading/ending `\n` together with whitespaces from the string slice of the raw code.